libr/util/unum.c

```diff
@@ -75,32 +75,37 @@ R_API void r_num_free(RNum *num) {
 	R_FREE (num);
 }
 
-#define KB (1024)
-#define MB (1024*KB)
-#define GB (1024*MB)
-#define TB (1024*GB)
+#define KB (1ULL<<10)
+#define MB (1ULL<<20)
+#define GB (1ULL<<30)
+#define TB (1ULL<<40)
 
+// TODO: Use SI system
+// https://metacpan.org/pod/Number::Bytes::Human
 R_API char *r_num_units(char *buf, ut64 num) {
 	char unit;
 	int tnum;
 	double fnum = num;
-	if (!buf) {
-		buf = malloc (32);
-		if (!buf) {
+#undef R_NUM_UNITS_BUFLEN
```

Comment:
i don't like too much the `#define` in the middle of the function

Reply:
yes but since you are also suggesting the "at least 32 bytes in size" in the documentation on top of `r_num_units` it would be better to say "at least R_NUM_UNITS_BUFLEN bytes in size" and it's very likely users of that API would want to allocate a buffer by using `char buf[R_NUM_UNITS_BUFLEN]`. By defining the constant here you are making it private, while it would probably fit better in r_num.h so that users of RNum could use it.

Reply:
you can use `const int unitsBuflen = 32`. no need for a define here

Reply:
I think @ret2libc is right. User could use R_NUM_UNITS_BUFLEN to allocate a buffer by using char `buf[R_NUM_UNITS_BUFLEN+1]` later.